internal/storage/leveldb.go

```diff
@@ -126,13 +166,20 @@ func (s *LevelDBStorage) SetTxStatus(queryID uint64, hash string, neutronHash st
 		return fmt.Errorf("failed to set tx txInfo: %w", err)
 	}
 
+	if processedTx != nil {
+		err = cacheProcessedTx(t, queryID, hash, *processedTx)
```

Comment:
why dereferencing `*processedTx`? you can make the `cacheProcessedTx` accept a pointer